source/Planet.cpp

```diff
@@ -446,7 +481,7 @@ bool Planet::IsInhabited() const
 // Check if this planet has a permanent shipyard.
 bool Planet::HasShipyard() const
 {
-	return !shipSales.empty();
+	return !shipSales.empty() && toAccessShipyard.Test();
```

Comment:
Empty conditions return zero by default. Is that the behavior we want here? (Or should no condition mean that access is granted?)

Reply:
Empty sets return true.

https://github.com/endless-sky/endless-sky/blob/2fbf66bfbac07a2193933b465aa01b1afd45082a/source/ConditionSet.cpp#L199-L208

https://github.com/endless-sky/endless-sky/blob/2fbf66bfbac07a2193933b465aa01b1afd45082a/source/ConditionSet.cpp#L367-L371

Otherwise missions with no `to offer` would never offer, and missions that don't specify `to fail` wouldn't need to check if the condition set is empty.
https://github.com/endless-sky/endless-sky/blob/2fbf66bfbac07a2193933b465aa01b1afd45082a/source/Mission.cpp#L919-L923

Reply:
Ok, I was briefly confused by the `return 0` in Evaluate(), but you're right, empty boolean conditions return true (because they have the AND operator by default).